lib/client/ipam_block.go

```diff
@@ -158,19 +159,31 @@ func (b *allocationBlock) release(addresses []cnet.IP) ([]cnet.IP, map[string]in
 
 	// Determine the ordinals that need to be released and the
 	// attributes that need to be cleaned up.
+	uniqueAddresses := set.New()
 	for _, ip := range addresses {
+		uniqueAddresses.Add(&ip)
```

Comment:
Hm, I'm not sure that using a pointer here will do the right thing.

For example, we could have two separate structures with the same value which wouldn't count as duplicates in this approach IIUC. 

I think we might need to de-dupe with a map, using a key which is a proxy for the value of the structure, which is really what we want to be unique.

e.g.

```
deduped := map[string]net.IP{}
for _, ip := range addresses {
    deduped[ip.String()] = ip
}

for _, ip := range deduped {
    // Existing logic
    ... 
}
```

WDYT?

Reply:
Yeah I think you’re correct. I actually implemented a String() -> MustParseIP() version but thought it would be more efficient with pointers. Will push the suggested version tonight.